engine/src/main/java/org/hibernate/search/engine/environment/bean/impl/BeanSource.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.environment.bean.impl;
+
+public enum BeanSource {
+
+	/**
+	 * The beans defined using
+	 * {@link org.hibernate.search.engine.environment.bean.spi.BeanConfigurer}s.
+	 */
+	CONFIGURATION,
+	/**
+	 * The bean manager, e.g. CDI, Spring, ...
+	 */
+	BEAN_MANAGER,
```

Comment:
While I agree this name is an approximation, we need to say *something* about the source in error messages, and I'd rather keep consistent wording everywhere.

Right now I feel that "bean manager" will be the clearest way to say "CDI/Spring" to users without saying it explicitly. And it's currently accurate since the only `BeanProvider` implementations we currently use are based on CDI/Spring.